### PR TITLE
Add dependencies for identity

### PR DESCRIPTION
**Pull request by Ben-Sheppard**
Resolves https://github.com/camunda-cloud/identity/issues/993

## What is the purpose of the change

Currently the dependencies listed for version 8.0 and next are for the IAM component. The Identity component replaced the IAM component from 8.0+ so these dependencies need to be updated to reflect the true state.

## Are there related marketing activities

Nope :)

## When should this change go live?

This change can go live whenever.

## PR Checklist

- [x] My changes apply to an already released version, and I have added them to the relevant `/versioned_docs` directory, or they are not for an already released version.
- [x] My changes apply to future versions, and I have added them to the main `/docs` directory, or they are not for future versions.
- [x] My changes require an [Engineering review](https://github.com/camunda/camunda-platform-docs/blob/main/howtos/documentation-guidelines.md#review-process), and I've assigned an engineering manager or tech lead as a reviewer, or my changes do not require an Engineering review.
- [x] My changes require a [technical writer review](https://github.com/camunda/camunda-platform-docs/blob/main/howtos/documentation-guidelines.md#review-process), and I've assigned @christinaausley as a reviewer, or my changes do not require a technical writer review.

### CODE REVIEW

File: docs/reference/dependencies.md [hunk 1]
```diff
@@ -622,309 +622,272 @@ import TabItem from "@theme/TabItem";
...
+- [com.squareup.okhttp3:logging-interceptor:4.9.3](https://square.github.io/okhttp/) (The Apache Software License, Version 2.0)
+- [com.squareup.okhttp3:okhttp:4.9.3](https://square.github.io/okhttp/) (The Apache Software License, Version 2.0)
+- [com.squareup.okio:okio:2.8.0](https://github.com/square/okio/) (The Apache Software License, Version 2.0)
+- [com.sun.activation:jakarta.activation:1.2.2](https://github.com/eclipse-ee4j/jaf/jakarta.activation) (EDL 1.0)
+- [com.sun.istack:istack-commons-runtime:3.0.12](https://projects.eclipse.org/projects/ee4j/istack-commons/istack-commons-runtime) (Eclipse Distribution License - v 1.0)
+- [com.sun.mail:jakarta.mail:1.6.7](http://eclipse-ee4j.github.io/mail/jakarta.mail) (EDL 1.0) (EPL 2.0) (GPL2 w/ CPE)
+- [commons-codec:commons-codec:1.15](https://commons.apache.org/proper/commons-codec/) (Apache License, Version 2.0)
+- [commons-io:commons-io:2.5](http://commons.apache.org/proper/commons-io/) (Apache License, Version 2.0)
+- [dev.failsafe:failsafe:3.3.0](https://failsafe.dev/failsafe) (Apache License, Version 2.0)
+- io.camunda:identity-sdk:8.0.0-SNAPSHOT (Unknown license)
```
Comment (Ben-Sheppard) [inline docs/reference/dependencies.md:813]: I'm not sure how we should handle this dependency on ourselves

Comment (dlavrenuek) [reply docs/reference/dependencies.md:813]: All sub dependencies of the SDK should also be included in this list, right? In this case we can just remove this entry.

Comment (Ben-Sheppard) [reply docs/reference/dependencies.md]: It appears that way yes, so I've removed this line (and the corresponding line in the versioned docs too)

**Review by akeller — DISMISSED** [review]
Approving, but please confirm with @dlavrenuek things like the dependencies are correct from an engineering perspective. Then you are good to merge!

**Comment by dlavrenuek** [comment]
@Ben-Sheppard there were some security updates in the last days and new dependency will be added in https://github.com/camunda-cloud/identity/pull/1135. We should be able to update this list and merge next Tuesday (04. Oct) shortly before the new minor release

**Comment by dlavrenuek** [comment]
@Ben-Sheppard I will 👍

**Review by akeller — APPROVED** [review]
👋 goodbye IAM 👋

**Pull request merged**